.github/workflows/inspektor-gadget.yml

```diff
@@ -78,100 +151,268 @@ jobs:
-            kubectl-gadget LICENSE
-          rm -rf $platform
-        done
-
-    - name: Build local-gadget for all platforms
+  test-basic:
+    name: Basic unit tests
+    # level: 0
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup go 1.17
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.17
+      id: go
+    - name: Cache go 1.17
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Cache debian packages
+      uses: actions/cache@v2
+      id: cache-debs
+      with:
+        path: "~/cache-debs"
+        key: cache-debs-libbpf-1:0.4.0-1ubuntu1-libseccomp-2.5.1-1ubuntu1~20.04.1
+    - name: Install debian packages
+      env:
+        CACHE_HIT: ${{steps.cache-debs.outputs.cache-hit}}
       run: |
-        make local-gadget-all
-
-        # Prepare assets for release and actions artifacts
-        for target in $(make list-local-gadget-targets); do
-          platform=$(echo $target|cut -d- -f3-4)
-          mkdir $platform
-          cp local-gadget-$platform $platform/local-gadget
-          cp LICENSE $platform/
-          tar --sort=name --owner=root:0 --group=root:0 \
-            -czf local-gadget-$platform.tar.gz -C $platform \
-            local-gadget LICENSE
-          rm -rf $platform
-        done
-
+          if [[ "$CACHE_HIT" == 'true' ]]; then
+            sudo cp --verbose --force --recursive ~/cache-debs/* /
+          else
+            sudo apt install -y software-properties-common
+            sudo add-apt-repository -y ppa:tuxinvader/kernel-build-tools
+            sudo apt-get update
+            sudo apt install -y libbpf-dev="$LIBBPF_VERSION" libseccomp-dev="$LIBSECCOMP_VERSION"
+            mkdir -p ~/cache-debs
+            sudo dpkg -L libbpf-dev libseccomp-dev | \
+                while IFS= read -r f; do \
+                    if test -f $f; then echo $f; fi; \
+                done | xargs cp --parents --target-directory ~/cache-debs/
+          fi
+    - name: Check out code
+      uses: actions/checkout@v1
     - name: Basic unit tests
       run: |
         make test
 
+  test-controller:
+    name: Unit tests for the controller
+    # level: 0
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup go 1.17
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.17
+      id: go
+    - name: Cache go 1.17
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v1
     - name: Unit tests for the controller
       run: |
         make controller-tests
 
+  test-local-gadget:
+    name: Unit tests for local-gadget
```

Comment:
![image](https://user-images.githubusercontent.com/16539896/151203885-0d12272b-e15d-47ad-bd14-bf9672c4ec95.png)

There seems to be some duplication of string `Unit tests for local-gadget`, which maybe could be avoided?

Reply:
I think this double printing is rather a problem within GitHub.
Indeed, all names are duplicated because the name of the job is the name of the main step within it.
So, it should be either printed all the times or never and not only sometimes.